third-party/realdds/doc/DDS ICD.md

```diff
@@ -0,0 +1,147 @@
+# realdds Topics
+
+There are 2 topic types in the realdds domain
+* **flexible** - Carries a buffer of data that can be JSON, CBOR or other user custom data
+See https://github.com/IntelRealSense/librealsense/blob/dds/third-party/realdds/include/realdds/topics/flexible/flexible.idl
+* **image** - A buffer of data with some extra fields that describe the image
+See https://github.com/IntelRealSense/librealsense/blob/dds/third-party/realdds/include/realdds/topics/image/image.idl
+
+There are 4 main topics, consisted of these 2 types.
+* **realdds/device-info** of type flexible - The server sends a sample of this topic to inform readers of a connected camera.
+  Some of the details included are the camera model and serial number, this information will be used in the other topic names to ease communication with verious cameras in the network.
+* **realdds/\<model\>/\<serial\>/notification** of type flexible - Notifications are sent from the camera server to the camera user.
+  Some notificaitons are being sent at discovery to convey the full device information, e.g streams, supported options, etc...
+  Other notifications are sent as needed, mostly as reply to a control message
+* **realdds/\<model\>/\<serial\>/control** of type flexible - Controls are commands from the camera user to the camera, e.g start streaming, set option value etc...
+* **realdds/\<model\>/\<serial\>/<stream name>** of type image - Frames of data of the appropriate stream from the camera server to the user.
+
+# Messages
+
+## device-info
+
+On the camera user side, `realdds::dds_device_watcher` can be used to listen to incoming device-info messages.
+When compiling librealsense with the `BUILD_WITH_DDS` flag the context automatically creates and uses a `realdds::dds_device_watcher` object to identify DDS connected cameras.
+
+The server side can use `realdds::dds_device_broadcaster` to add or remove cameras from the DDS network.
+When a camera is added to the network the broadcaster will crerate a dds writer for it. If there is a **realdds/device-info** reader that mathces the broadcaster writer, topic sample will be sent with the appropriate device details.
+When a camera is removed from the network the broadcaster will close the writer, thus informing possible readers that the camera is disconnected.
+
+### device-info message format example
+
+device-info uses the **flexible** type with a JSON format of the following structure
+
```

Comment:
This is an example not the structure.
We should add a link to the structure and then show this example

Reply:
We don't have a structure defined for every JSON message, we use JSON exactly for this flexibility and not defining a type.
Specifically, the device-info used to have another type (another IDL) and I left a helper structure to minimize code changes, but using this structure is not necessary.

Reply:
I used a complete example because just listing the JSON map keys without values felt less informative. Because it is a free definition I wanted to show where we expect int, float, string or even a list...